Sending email functionality

YashRavipati1 opened the pull request:
## Tracking Info

Resolves #69 

## Changes

<!-- What changes did you make? -->

Implemented functionality for automated emails (backend and frontend API service)

## Testing

<!-- How did you confirm your changes worked? -->

Created a custom button to send a pre-set automated email to a certain email address every time it was pressed.

## Confirmation of Change

<!-- Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change. -->

<img width="925" alt="Screenshot 2024-05-07 at 2 19 47 AM" src="https://github.com/TritonSE/USHS-Housing-Portal/assets/63521936/0e8ca6af-8dbb-45ca-bf02-152178679c02">

YashRavipati1 committed:
Sending email functionality

YashRavipati1 committed:
Sending email functionality

YashRavipati1 committed:
Sending email functionality

YashRavipati1 committed:
Sending email functionality

YashRavipati1 committed:
Sending email functionality
p 16008e0d55b23858d962445c6861d4b88fcf4c6d

YashRavipati1 committed:
Sending email functionality

Azhou2023 committed:
Resize Images (#96)

## Tracking Info

Resolves #92 

## Changes

Added resize images extension in Firebase automatically resize all
uploaded images to 600x400. Changed default thumbnail/carousel and made
carousel arrows more clear.

## Testing

Ensured that all uploaded images were correctly downscaled when
uploaded; observed difference in file size before and after resizing
(made files 20x smaller in some cases). Ensured that uploading from the
app with this extension was smooth and everything scaled correclty.

## Confirmation of Change

<img width="1512" alt="Screenshot 2024-05-06 at 2 41 13 PM"
src="https://github.com/TritonSE/USHS-Housing-Portal/assets/57163201/413fee20-6497-4971-b22e-c65be84dce6c">
<img width="337" alt="Screenshot 2024-05-06 at 2 41 26 PM"
src="https://github.com/TritonSE/USHS-Housing-Portal/assets/57163201/be984d32-6d0c-4450-8c08-cded0e998f8f">
<img width="1512" alt="Screenshot 2024-05-06 at 2 42 05 PM"
src="https://github.com/TritonSE/USHS-Housing-Portal/assets/57163201/6e6a1260-9ffe-47a1-955d-014bccd1e6d3">

---------

petabite reviewed (commented):
Do we need a route to send emails? Since all the events that trigger an email are processed in the backend, I was thinking we can just call the backend email service's `sendEmail` function wherever we need to send emails. That way we can save a round trip from the frontend to backend. Thoughts?

RamtinTJB committed:
V2 filters frontend (#90)

## Tracking Info

Resolves parts of #80 (Doesn't send the request to the backend on click)

## Changes

- Refactored old components
- Deleted old unused filter components
- Wrote reusable components for checkbox filters and radio button
filters
- Made the NavBar sticky and implemented the frontend for the filter
panel

## Testing

Since the "Apply" button doesn't actually send the request to the
backend, the only way to test it to make sure it visually looks the same
as the designs.

## Confirmation of Change


![image](https://github.com/TritonSE/USHS-Housing-Portal/assets/69605985/0431bdc9-24a2-41d6-818a-0f9cf14a6652)

---------

Co-authored-by: Philip Zhang <code@philipzha.ng>

PoliteUnicorn committed:
Profile Page: Manage Permissions (#91)

## Tracking Info

Resolves #76
Resolves #83 
Resolves #67 

## Changes
I implemented the /users/:id/demote route which will demote a user from
a housing locator to simply a user. I added the functions
demoteUserHandler, a route for demote user, and demoteUser.

For issue 83, I also created the Housing Locator table and have a trash
icon next to each housing locator. When it is clicked, that housing
locator will be demoted.

There's also a pop up when the trash can is clicked.

## Testing
I tested the route so that when the delete button calls the demote user
route, the user is removed and their status is changed on MongoDB.


## Confirmation of Change
When the delete button next to the housing locator's name is clicked,
the user is removed and their status is changed on MongoDB.

When I click on the trash button by my own name - 
<img width="1440" alt="Screenshot 2024-04-23 at 6 39 52 PM"
src="https://github.com/TritonSE/USHS-Housing-Portal/assets/35088854/e50b6d83-4763-4202-a2fa-63399aee3b88">

Then, this is my view after I am no longer a housing locator 
<img width="1184" alt="Screenshot 2024-04-23 at 6 40 13 PM"
src="https://github.com/TritonSE/USHS-Housing-Portal/assets/35088854/f56e0a8b-1152-4945-9887-114578a7cd41">

Finally, this is my status after it is updated in MongoDB and I am no
longer a housing locator
<img width="686" alt="Screenshot 2024-04-23 at 6 40 33 PM"
src="https://github.com/TritonSE/USHS-Housing-Portal/assets/35088854/d33225a9-ed47-409c-a988-3134e8ab0728">

<img width="1032" alt="Screenshot 2024-04-24 at 3 34 09 PM"
src="https://github.com/TritonSE/USHS-Housing-Portal/assets/35088854/55791c75-2e2e-40b5-b52b-77480331f054">
<img width="865" alt="Screenshot 2024-04-24 at 3 34 15 PM"
src="https://github.com/TritonSE/USHS-Housing-Portal/assets/35088854/accdf978-921a-401e-93c5-56f6ef200219">

---------

Co-authored-by: Philip Zhang <code@philipzha.ng>

nbpillai committed:
Renter Candidate Page (#95)

## Tracking Info

Resolves #84 
Resolves #66 

## Changes

I implemented a get route in the backend to get information about a
renter candidate.

Extract `Table` component for reuse and refactor code in `ReferralTable`
to use it.

## Testing

I tested the route on Postman and implemented the frontend on the renter
candidate page to display the information.



## Confirmation of Change


![image](https://github.com/TritonSE/USHS-Housing-Portal/assets/24444266/2995c3c0-c6fa-4224-a3c8-4663ffa90454)

---------

Co-authored-by: Philip Zhang <code@philipzha.ng>

YashRavipati1 committed:
Removing email route

YashRavipati1 committed:
Removing email route

YashRavipati1 committed:
Removing email route

YashRavipati1 committed:
hooking email functionality

YashRavipati1 committed:
hooking email functionality

YashRavipati1 committed:
Merge branch 'feature/Yash/EmailSending' of github.com:TritonSE/USHS-Housing-Portal into feature/Yash/EmailSending

YashRavipati1 committed:
Demote User

YashRavipati1 committed:
Merge branch 'feature/Yash/EmailSending' of github.com:TritonSE/USHS-Housing-Portal into feature/Yash/EmailSending

YashRavipati1 committed:
Merge branch 'feature/Yash/EmailSending' of github.com:TritonSE/USHS-Housing-Portal into feature/Yash/EmailSending

YashRavipati1 committed:
Revert "Demote User"

This reverts commit f22c124ef66367837ee0ee6a218f156e218ca167.

YashRavipati1 committed:
Demote User Email

YashRavipati1 committed:
Removing route for email

YashRavipati1 committed:
Revert "Sending email functionality"

This reverts commit 4dd63e7c536d1c8a76e3027309747b281b251e61.

YashRavipati1 committed:
Revert "Resize Images (#96)"

This reverts commit 91d681e03caaf06daeb98dd565dc3b053c1aaf48.

YashRavipati1 committed:
Revert "V2 filters frontend (#90)"

This reverts commit 02ad2f7d90b5c11bef3d7b5bbc894b013d204660.

YashRavipati1 committed:
Revert "Profile Page: Manage Permissions (#91)"

This reverts commit a805420164271380bf5934dc5fa333f033facd13.

YashRavipati1 committed:
Revert "Renter Candidate Page (#95)"

This reverts commit 4b5fa7d00aee9f9d7a8e08f4d6e3037c6037fb95.

YashRavipati1 committed:
demoting user

YashRavipati1 committed:
Cleaning up/removing frontend routes

petabite commented on frontend/package-lock.json:
Can we delete this file? Looks like you maybe accidentally ran `npm install`

petabite commented on backend/src/services/email.ts:
```diff
@@ -0,0 +1,26 @@
+  });
+
+  const mailData = {
+    from: "dummyushsacc@gmail.com", // sender address
```
Is this the same as `process.env.GMAILUSER`?

petabite commented on backend/src/services/referral.ts:
```diff
@@ -33,11 +36,39 @@ export async function editReferral(
+  const Ref = await ReferralModel.findById(id);
+  const updateHL = Ref?.assignedHousingLocator?.toString() !== assignedHousingLocatorId;
+  const setLeased =
+    status === "leased" &&
```
```suggestion
    status === "Leased" &&
```
Should this be `"Leased"` if we are expecting all statuses to start with a capital letter?

petabite commented on backend/src/services/referral.ts:
```diff
@@ -33,11 +36,39 @@ export async function editReferral(
+      );
+    }
+  }
+  if (setLeased) {
+    if (HL !== null) {
+      await sendEmail(
+        HL?.email,
+        "Referral Update",
+        `Your referral has been marked as leased. Please login to the portal to view the update.`,
+      );
+    }
+  }
```
> - When referral status(es) for unit is set to leased → notify all HLs associated with the ACTIVE referrals for the unit
>     - inactive referral statuses = [ “Denied”, “Canceled” ]

Oops maybe I wasn't too clear about this. When a referral gets set to "Leased", the unit for that referral is leased, so we want to notify all _other_ referrals for that unit that the unit is not available anymore. This currently looks like it just notifies the HL for the current referral.

petabite reviewed (changes requested):
Looking great so far! Just a couple things:

YashRavipati1 committed:
Fixing referall leased emails

YashRavipati1 committed:
assignedHousingLocator instead of renterCandidate

petabite committed:
build failure

petabite reviewed (approved):
💯 🔥

Azhou2023 reviewed (approved):
LGTM!
